plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/settings/CodeWhispererConfigurable.kt

```diff
@@ -87,6 +89,55 @@ class CodeWhispererConfigurable(private val project: Project) :
             }
         }
 
+        group(message("aws.settings.codewhisperer.group.q_chat")) {
+            row {
+                checkBox(message("aws.settings.codewhisperer.project_context")).apply {
+                    connect.subscribe(
+                        ToolkitConnectionManagerListener.TOPIC,
+                        object : ToolkitConnectionManagerListener {
+                            override fun activeConnectionChanged(newConnection: ToolkitConnection?) {
+                                enabled(isCodeWhispererEnabled(project))
+                            }
+                        }
+                    )
+                    enabled(invoke)
+                    bindSelected(codeWhispererSettings::isProjectContextEnabled, codeWhispererSettings::toggleProjectContextEnabled)
+                }.comment(message("aws.settings.codewhisperer.project_context.tooltip"))
+            }
+
+            row {
+                intTextField(
+                    range = IntRange(0, 1000)
```

Comment:
1000 threads seems like too many to allow

Reply:
there is server side validation, but yes I can change it to a smaller number here too